examples/generative/customer_lifetime_value.py

```diff
@@ -0,0 +1,574 @@
+    decoder_initial = RepeatVector(
+        output_sequence_length, name="repeat_merged_features"
+    )(merged_features)
+
+    decoder_lstm = LSTM(
+        units=64,
+        return_sequences=True,
+        recurrent_regularizer=regularizers.L1L2(l1=1e-5, l2=1e-4),
+    )(decoder_initial)
+
+    # Output Dense layer
+    output = Dense(units=1, activation="linear", name="output_dense")(decoder_lstm)
+
+    model = Model(
+        inputs=[temporal_inputs, trend_inputs, country_inputs], outputs=output
+    )
+
+    model.compile(
+        optimizer=tf.keras.optimizers.Adam(learning_rate=0.001),
+        loss="mse",
+        metrics=["mse"],
+    )
+
+    return model
+
+
+# Create the hybrid model
+model = build_hybrid_model(
+    input_sequence_length=6,
+    output_sequence_length=6,
+    num_countries=len(np.unique(train_data["static_features"])) + 1,
+    d_model=8,
+    num_heads=4,
+)
+
+# Configure StringLookup
+label_encoder = StringLookup(output_mode="one_hot", num_oov_indices=1)
+
+# Adapt and encode
+label_encoder.adapt(train_data["static_features"])
+
+train_static_encoded = label_encoder(train_data["static_features"])
+val_static_encoded = label_encoder(val_data["static_features"])
+test_static_encoded = label_encoder(test_data["static_features"])
+
+# Convert sequences with proper type casting
+X_train_seq = np.asarray(train_data["trend_sequences"]).astype(np.float32)
+X_val_seq = np.asarray(val_data["trend_sequences"]).astype(np.float32)
+X_train_temporal = np.asarray(train_data["temporal_sequences"]).astype(np.float32)
+X_val_temporal = np.asarray(val_data["temporal_sequences"]).astype(np.float32)
+
+# Define callbacks with proper monitoring strategy
+callbacks = [
+    keras.callbacks.EarlyStopping(
+        monitor="val_loss",
+        patience=10,
+        restore_best_weights=True,  # Important for best model preservation
+        mode="min",
+    ),
+    keras.callbacks.ReduceLROnPlateau(
+        monitor="val_loss", factor=0.2, patience=5, min_lr=1e-6, mode="min", verbose=1
+    ),
+]
+
+train_outputs = train_data["output_sequences"].astype(np.float32)
+val_outputs = val_data["output_sequences"].astype(np.float32)
+test_output = test_data["output_sequences"].astype(np.float32)
+# Training setup
+
+history = model.fit(
+    [X_train_temporal, X_train_seq, train_static_encoded],
+    train_outputs,
+    validation_data=(
+        [X_val_temporal, X_val_seq, val_static_encoded],
+        val_data["output_sequences"].astype(np.float32),
+    ),
+    epochs=20,
+    batch_size=32,
+    callbacks=[
+        tf.keras.callbacks.EarlyStopping(
+            monitor="val_loss", patience=10, restore_best_weights=True
+        )
+    ],
+)
+
+# Make predictions
+predictions = model.predict(
+    [
+        test_data["temporal_sequences"].astype(np.float32),
+        test_data["trend_sequences"].astype(np.float32),
+        test_static_encoded,
+    ]
+)
+
+# Calculate the predictions
+predictions = tf.squeeze(predictions)
```

Comment:
Use np.squeeze (this array is a numpy array)

Reply:
Done!